core/src/test/java/org/apache/calcite/test/RexProgramBuilderBase.java

```diff
@@ -316,6 +317,14 @@ protected RexNode add(RexNode n1, RexNode n2) {
     return rexBuilder.makeCall(SqlStdOperatorTable.PLUS, n1, n2);
   }
 
+  protected RexNode item(RexInputRef inputRef, int index) {
+    RexNode rexNode = rexBuilder.makeCall(
+        SqlStdOperatorTable.ITEM,
+        inputRef,
```

Comment:
The `ITEM` operator can take both INTEGER and VARCHAR as 1th arguments, if the left record is a `MAP` type, the 1th argument can be type of the `MAP` key type, so only a index here is not a complete implementation.

Reply:
The scope of the tests keeps increasing :)   I modified the second parameter to take a `RexNode literal` instead and added a unit test for the map case.